src/atlas/grid/CubedSphereGrid.h

```diff
@@ -0,0 +1,81 @@
+/*
+ * (C) Copyright 2020 UCAR
+ *
+ * This software is licensed under the terms of the Apache Licence Version 2.0
+ * which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+#pragma once
+
+#include <functional>
+#include <initializer_list>
+#include <string>
+
+#include "atlas/grid/Grid.h"
+#include "atlas/grid/detail/grid/CubedSphere.h"
+
+namespace atlas {
+
+//---------------------------------------------------------------------------------------------------------------------
+class CubedSphereGrid;
+class EquiDistCubedSphereGrid;
+class EquiAnglCubedSphereGrid;
+
+/*
+                                             Grid
+                                               |
+                                    +----------+----------+------------------------+
+                                    |                     |                        |
+                             StructuredGrid        UnstructuredGrid           CubedSphere
+                                    |                                              |
+               +--------------------+-----------------------+               +------+------+
+               |                    |                       |               |             |
+          ReducedGrid          GaussianGrid            RegularGrid       EquiDist      EquiAngl
+               |                 |     |                 |     |
+               +--------+--------+     +--------+--------+     +-----+
+                        |                       |                    |
+               ReducedGaussianGrid     RegularGaussianGrid    RegularLonLatGrid
+*/
```

Comment:
@danholdaway 
Then wouldn't `CubedSphere` be inheriting from `StructuredGrid` as opposed to `Grid`
Or is `StructuredGrid` incapable of defining the `tile` axis?

Reply:
Yeah, the tile dimension throws in a wrinkle. Think of the image "2D x and y" it would have to represented with a continuous 2D array, resulting in many ghost nodes a very messy grid iterator. Having the tile dimension makes everything a lot easier.

Reply:
@aerorahul The `atlas::StructuredGrid` itself is a "one-tile" grid (regional rectangular grids, likely with projections, or global longitude-latitude grids), where points are organized in rows (parallels in y-direction) which are itself not necessarily equidistant. Points on one parallel are equidistant (along x-direction). A cubed sphere grid could not fit in this picture as there are 6 structured tiles.
The best solution would be to create a family of tiled grids, and the cubed sphere, and possibly other grids could fit in this category. But as long as we only have one candidate implementation, I would refrain from already making a TiledGrid base class for this.

Reply:
@wdeconinck I understand. While `CubedSphere` is a `StructuredGrid`, it is infact a collection of sorts of 6 `StructuredGrid` objects.  I imagine a `TriPolarGrid` will be its own `Grid`.  In essence `StructuredGrid` is really a `RectilinearGrid`.  I guess its just a name.